tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsWriter.cs

```diff
@@ -48,6 +55,30 @@ internal class DataStreamsWriter
             period: bucketDurationMs);
     }
 
+    /// <summary>
+    /// Public for testing only
+    /// </summary>
+    public event EventHandler<EventArgs>? FlushComplete;
+
+    private enum SupportState
+    {
+        Unknown,
+        Supported,
+        Unsupported
+    }
+
+    public static DataStreamsWriter Create(
+        ImmutableTracerSettings settings,
+        IDiscoveryService discoveryService,
+        string defaultServiceName)
+        => new DataStreamsWriter(
+            new DataStreamsAggregator(
+                new DataStreamsMessagePackFormatter(settings.Environment, defaultServiceName),
+                bucketDurationMs: DataStreamsConstants.DefaultBucketDurationMs),
+            new DataStreamsApi(DataStreamsTransportStrategy.GetAgentIntakeFactory(settings.Exporter)),
+            bucketDurationMs: DataStreamsConstants.DefaultBucketDurationMs,
+            discoveryService);
+
     public void Add(in StatsPoint point)
     {
         if (_buffer.TryEnqueue(point))
```

Comment:
I think you mentioned this in another PR comment, but when we know `_isSupported == SupportState.Unsupported`, we should consider skipping the add. Since `_isSupported` will be written very few times and read a lot, we should prioritize faster reads, but I don't honestly know at the moment which primitives are the best to optimize for that

Reply:
Added a read of `_isSupported` to not add the points when explicitly unsupported in https://github.com/DataDog/dd-trace-dotnet/pull/3220/commits/583cb56c091271495f028a404cd2f10cb238dd05. `Volatile.Read()` should be good enough I think.